src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1914 @@
+                if (r < 0)
+                        log_warning_errno(r, "Cannot emit JobRemoved message, ignoring: %m");
+        }
+
+        if (j->dbus_msg && j->complete_cb) {
+                if (sd_bus_error_is_set(&error)) {
+                        log_warning("Bus error occurred, ignoring callback for job: %s", error.message);
+                        sd_bus_reply_method_error(j->dbus_msg, &error);
+                } else {
+                        r = j->complete_cb(j->dbus_msg, j, json, &error);
+                        if (r < 0) {
+                                log_warning_errno(r, "Error during execution of job callback: %s", bus_error_message(&error, r));
+                                sd_bus_reply_method_errno(j->dbus_msg, r, &error);
+                        }
+                }
+        }
+
+        if (j->manager && hashmap_isempty(j->manager->jobs))
+                manager_became_idle(j->manager);
+
+        job_free(j);
+
+        return 0;
+}
+
+static inline const char* sysupdate_binary_path(void) {
+        return secure_getenv("SYSTEMD_SYSUPDATE_PATH") ?: SYSTEMD_SYSUPDATE_PATH;
+}
+
+static int target_get_argument(Target *t, char **ret) {
+        _cleanup_free_ char *target_arg = NULL;
+
+        assert(t);
+        assert(ret);
+
+        if (t->class != TARGET_HOST) {
+                if (t->class == TARGET_COMPONENT)
+                        target_arg = strjoin("--component=", t->name);
+                else if (IN_SET(t->image_type, IMAGE_DIRECTORY, IMAGE_SUBVOLUME))
+                        target_arg = strjoin("--root=", t->path);
+                else if (IN_SET(t->image_type, IMAGE_RAW, IMAGE_BLOCK))
+                        target_arg = strjoin("--image=", t->path);
+                else
+                        assert_not_reached();
+                if (!target_arg)
+                        return -ENOMEM;
+        }
+
+        *ret = TAKE_PTR(target_arg);
+        return 0;
+}
+
+static int job_start(Job *j) {
+        _cleanup_close_ int stdout_fd = -EBADF;
+        _cleanup_(pidref_done_sigkill_wait) PidRef pid = PIDREF_NULL;
+        int r;
+
+        assert(j);
+
+        if (IN_SET(j->type, JOB_UPDATE, JOB_VACUUM) && j->target->busy) {
+                log_info("Target %s busy, ignoring job.", j->target->name);
+                return -EBUSY;
+        }
+
+        stdout_fd = memfd_new("sysupdate-stdout");
+        if (stdout_fd < 0)
+                return log_error_errno(errno, "Failed to create memfd: %m");
+
+        r = pidref_safe_fork_full("(sd-sysupdate)",
+                                  (int[]) { -EBADF, stdout_fd, STDERR_FILENO }, NULL, 0,
+                                  FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG_SIGTERM|
+                                  FORK_REARRANGE_STDIO|FORK_LOG|FORK_REOPEN_LOG, &pid);
+        if (r < 0)
+                return log_error_errno(r, "Failed to fork (sd-sysupdate): %m");
```

Comment:
you specified FORK_LOG as flag, which means pidref_safe_fork_full() will already log for you, hence just do `return r;` here, otherwise you'll log twice